python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ATGetStdFlatDataset"]
+
+import numpy as np
+
+from lsst.ts import salobj
+from lsst.ts import scriptqueue
+
+import SALPY_ATCamera
+import SALPY_ATSpectrograph
+
+
+class ATGetStdFlatDataset(scriptqueue.BaseScript):
+    """Implement script to get sensor characterization data. The definition is spelled out in
+    https://jira.lsstcorp.org/browse/CAP-203 and https://jira.lsstcorp.org/browse/CAP-206.
+
+    Basically, this script will:
+
+    1 - Take a set `nd` (default=10) dark images (shutter closed) with `td` (default=400s) exposure time
+    2 - Take a set of `nb` (default=10) bias
+    3 - Flat field data:
+        - Take a set of pairs of flat fields at a set of approximately logarithmically spaced intensity
+        levels starting at 500 DN and increasing by a factor of 2 (i.e. 500, 1000, 2000, ...).
+        The exact levels are not important, but they must be well known; both the flux level and
+        shutter time must be well measured (if shutter is opened before/closed after the lamp
+        is turned on then the shutter time need not be well measured).
+        - Take a set of `nb` biases after the entire flat sequence is complete
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+    """
+
+    def __init__(self, index,
+                 descr="Take Flat field sensor characterization data."):
+
+        super().__init__(index=index,
+                         descr=descr,
+                         remotes_dict=dict(atcam=salobj.Remote(SALPY_ATCamera,
+                                                               include=["takeImages",
+                                                                        "endReadout"]),
+                                           atspec=salobj.Remote(SALPY_ATSpectrograph,
+                                                                include=["changeFilter",
+                                                                         "changeDisperser",
+                                                                         "moveLinearStage"])))
+
+        # Had to increase this timeout to accommodate some issue with the camera.
+        self.cmd_timeout = 120.
+        self.read_out_time = 2.
+        self.n_dark = 10
+        self.t_dark = 400.
+        self.n_bias = 10
+        self.n_flat = 2
+        self.flat_base_exptime = 0.5
+        self.flat_dn_range = np.array([1., 2., 4., 8., 16., 32., 64., 128.])
+        self.filter = None
+        self.grating = None
+        self.linear_stage = None
```

Comment:
These lines duplicate the code in `configure`. Consider deleting these and using the values above as the default values for the arguments in `configure` (except the array should be a tuple to avoid mutable arguments). Then check the value provided and, if acceptable, just set the value. In other words:
```
configure(self, n_dark=10, ...):
    if n_dark <= 0:
        raise...
    self.n_dark = n_dark
```
with no need to check for None